invenio_rdm_records/resources/serializers/schemas.py

```diff
@@ -79,3 +86,17 @@ def get_contributors(self, obj):
         return [
             c["person_or_org"]["name"] for c in obj["metadata"].get("contributors", [])
         ] or missing
+
+    @post_dump(pass_original=True)
+    def post_dump_pipeline(self, data, original, many, **kwargs):
+        """Defines sequence of post dump steps."""
+        result = deepcopy(data)
+        for processor in self.processors:
+            if not hasattr(processor, "post_dump"):
+                raise TypeError("Post process processors must be callables")
```

Comment:
I would instead make a check in the `__init__` like:

```python
    def __init__(self, processors=None, **kwargs):
        super().__init__()
        assert all(isinstance(p, DumpProcessorMixin) for p in processors)
        self.processors = processors or []
```

Then by default I would make the `DumpProcessorMixin` implement an "identity" `post_dump`/`pre_dump` (i.e. `lambda x: x`)